Switch to alabaster theme

emlys opened the pull request.

jagoldstein reviewed (changes requested):
I like this theme much better than the current custom one, but I notice some formatting problems that should be addressed before putting this in production:
- margins are too wide
- odd line breaks with excessive word splitting (including in table column headers)
- no TOC within each model section

emlys commented:
@jagoldstein I think I've brought back the table of contents within model sections. Could you maybe attach a screenshot showing the other two points? Thanks

jagoldstein commented:
@emlys please see the screenshot here from Chrome on Windows. Two red lines show the width of the margins relative to the content. Blue marks circle some of the many seemingly excessive examples of word splitting.
![alabasterThemeUG](https://user-images.githubusercontent.com/18270215/199364329-d8b7970d-7cc1-4a60-8753-2802da262d1f.png)

phargogh reviewed (changes requested):
Just a couple comments on RST/python changes that I think should be pretty minor.

In general, the Alabaster theme is fine by me!  I kind of like our current theme, but appreciate that translations are a concern and have no problem with updating to a more modern theme.

Style-wise, here are a few minor things I noticed in the preview:
* Longer titles in the sidebar are wrapped to the next line (e.g. "InVEST Scripting Guide and API"), which can be a little frustrating to read
* The sidebar has only a very slight visual distinction from the body of the text.

jagoldstein commented on source/ndr.rst:
```diff
@@ -1,8 +1,8 @@
-****************************
-NDR: Nutrient Delivery Ratio
-****************************
+***********************
+Nutrient Delivery Ratio
+***********************
```
I totally search by "SDR" and "NDR" and also wish there was an "SWY" title!

dcdenu4 commented:
Hey @emlys, thanks for looking into this. James mentioned that part of the motivation to do this was translation related? Could you mention what those considerations are in the PR summary? 

>Remove the "Supporting & Final Ecosystem Services" vs. "Additional Tools" sections of the index. Alphabetize the list of models. And change some chapter titles to match the model title.

I feel like there is some history here with those two distinct sections. If you don't mind, I think it would be worth linking this new UG theme and changes in Slack and tagging @lmandle , @adrianvogl , and @newtpatrol. The software team has become the maintainer of the Users Guide over the years, but the UG should really be driven from the science / communication side (IMO), so I just want to make sure we're not making changes in a vacuum.

Sorry if this has already been hashed out and I missed it since I've been out!

dcdenu4 commented on source/globio.rst:
```diff
@@ -1,5 +1,5 @@
-InVEST GLOBIO Model
-===================
+GLOBIO
```
I wonder if InVEST GLOBIO is a good distinction from THE GLOBIO model? Just to reinforce this is a modified version of GLOBIO and not a repackage of that model.

dcdenu4 commented on source/wind_energy.rst:
```diff
@@ -1,8 +1,8 @@
-*******************************
-Offshore Wind Energy Production
-*******************************
+**********************
+Wind Energy Production
+**********************
```
I agree and think that we should cast a wider net of reviewers for some of these changes. I mentioned it in a comment before but there's a lot of history with the UG and how things ended up being labeled. I'd want to make sure we got at least Lisa in here to review, if not others.

@rmgriffin

rmgriffin replied:
As set up it is definitely an offshore wind model. I'm pretty sure it could be coerced to do terrestrial wind modeling, though I don't think it would be a satisfying experience. Are there other changes to this model, or others like wave energy or valuation text generally that need review?

dcdenu4 replied:
Thanks @rmgriffin, I don't think this PR is touching any substance other than the Theme and trying to have the UG reflect what people find in the UI. We will certainly tag you for wind, wave, and valuation in general if it comes up in the future!

emlys commented:
I didn't realize the title and TOC changes would be so controversial so I've removed them entirely.

phargogh reviewed (approved):
I'm fine with switching to alabaster and putting off theme tweaks until later.

davemfish commented:
> The purpose of this PR is to get the free, automatic translation that comes with the standard themes. I agree with most of the style issues people have pointed out, but they're a bit far removed from the scope of the issue I'm trying to address. Would it be ok to save that for #97?

I agree it's nice to not get bogged down in too many details that are tangential to the goal. But I think it's worth making sure we avoid making the UG harder to use, even temporarily. To that end, I think these items are worth addressing if we can:

>* Longer titles in the sidebar are wrapped to the next line (e.g. "InVEST Scripting Guide and API"), which can be a little frustrating to read
* The sidebar has only a very slight visual distinction from the body of the text.

I do like how all the model links are visible in the sidebar! (but you may have changed it back already, which is fine too).

But @emlys , if you do want to get on with the real task at hand and ignore all these details for now, what about merging this into a feature branch and then coming back to do the theme work later?

dcdenu4 commented:
>I didn't realize the title and TOC changes would be so controversial so I've removed them entirely. - Emily

Sorry! This is me projecting / anticipating the larger NatCap reaction from previous UG changes. Maybe it's not as controversial anymore as it once was.

>The purpose of this PR is to get the free, automatic translation that comes with the standard themes. - Emily

Ah, that is important, thanks.

>what about merging this into a feature branch and then coming back to do the theme work later? - Dave

This is what I was thinking as well.

dcdenu4 commented:
Hey @emlys, I completely missed the discussion you all were having in #94, sorry about that!

emlys commented:
The sidebar structure is now nearly the same as it was before:
<img width="717" alt="image" src="https://user-images.githubusercontent.com/43770515/200048902-4e9e434f-1b5f-4511-9c12-bd4f75cdcbbb.png">

<img width="1017" alt="image" src="https://user-images.githubusercontent.com/43770515/200048826-fe3c294a-2fff-483f-ba3d-c6ece10ee01d.png">

dcdenu4 reviewed (changes requested):
Hey @emlys, thanks for the extra effort in formatting! I just noticed one thing when checking out the artifact. I had made some changes to the citation / contributors over in `main` which included adding a `custom.css` file to `themes\natcapUG\static\` (https://github.com/natcap/invest.users-guide/blob/86aa472fc1b87c4b72fced5063bf7c23544430c6/source/conf.py#L108). 

I think removing `html_theme_path` in `conf.py` might make it so that css can't be found. I also noticed the updated citation and contributions weren't in the artifact build...

emlys commented:
Thanks for catching that @dcdenu4!
- I moved `custom.css` out of the themes directory which no longer exists
- I used `html_static_path` instead of `html_css_path` because the [alabaster docs](https://alabaster.readthedocs.io/en/latest/customization.html#custom-stylesheet) recommended it
- I changed the number of contributor columns from 5 to 3 so that names would fit on one line in the new theme

dcdenu4 reviewed (approved):
Thanks Emily!

jagoldstein commented:
@emlys I'm liking the new look, but would it be possible to lock the TOC as a panel on the left so that it doesn't scroll out of view as users move down a page?

emlys commented:
> @emlys I'm liking the new look, but would it be possible to lock the TOC as a panel on the left so that it doesn't scroll out of view as users move down a page?

good idea @jagoldstein, I've moved this over to #97
